fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/directory/FDBDirectoryLockFactory.java

```diff
@@ -181,6 +187,20 @@ private void fileLockCheckNewLock(byte[] val, long nowMillis) {
             }
         }
 
+        private CompletableFuture<Void> ensureValidFileLock(final FDBRecordContext aContext) {
+            final long nowMillis = System.currentTimeMillis();
+            return aContext.ensureActive().get(fileLockKey)
+                    .thenAccept(val -> {
+                        synchronized (fileLockSetLock) {
+                            if (val != null) {
+                                this.timeStampMillis = nowMillis;
+                                final byte[] value = fileLockValue();
+                                aContext.ensureActive().set(fileLockKey, value);
+                            }
+                        }
+                    });
```

Comment:
IIUIC, at this path the assumption is that we own the lock since `closed == false`. Hence we should: 
1. require a non null
2. assert that the lock matches `selfStampUuid`. 


One easy way to do it would be to return `fileLockSet(true, aContext)`, I think.

Reply:
`fileLockSet` throws exceptions when `val` is `null`

Reply:
The close path `fileLockClearFlushAndClose` clears the lock first then does flush/commit, then assigns close = true. val is always null because of this.
https://github.com/FoundationDB/fdb-record-layer/blob/main/fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/directory/FDBDirectoryLockFactory.java#L207-L211

Reply:
Can we change the refereed code to something like: 
```
try {
            closed = true; // prevent lock stamp update
            agilityContext.flush();
} catch (ex) {
            closed = false;  // allow close retry
            throw ex;
}
```
?

Reply:
Take a look now

Reply:
Nice! 
So now can we call `fileLockSet(true, aContext)` that will require valid uuid if not closed?

Reply:
It makes it the possible now (look at the latest change) but I couldn't find any tests making this call.